functions/Test-DbaServerName.ps1

```diff
@@ -0,0 +1,195 @@
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Test-DbaServerName
+
+.EXAMPLE
+Test-DbaServerName -SqlServer sqlserver2014a
+
+Returns ServerInstanceName, SqlServerName, IsEqual and RenameRequired for sqlserver2014a.
+
+.EXAMPLE   
+Test-DbaServerName -SqlServer sqlserver2014a, sql2016
+
+Returns ServerInstanceName, SqlServerName, IsEqual and RenameRequired for sqlserver2014a and sql2016.
+
+.EXAMPLE   
+Test-DbaServerName -SqlServer sqlserver2014a, sql2016 -Detailed
+
+Returns ServerInstanceName, SqlServerName, IsEqual and RenameRequired for sqlserver2014a and sql2016.
+	
+If a Rename is required, it will also show Updatable, and Reasons if the servername is not updatable.
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[PsCredential]$Credential,
+		[switch]$Detailed,
+		[switch]$NoWarning
+	)
+	
+	BEGIN
+	{
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+		$servercount++
+		
+		foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $Credential
+			}
+			catch
+			{
+				if ($servercount -eq 1)
+				{
+					throw $_
+				}
+				else
+				{
+					Write-Warning "Can't connect to $servername. Moving on."
+					Continue
+				}
+			}
+			
+			
+			if ($server.isClustered)
+			{
+				Write-Warning "$servername is a cluster. Renaming clusters is not supported by Microsoft."
+			}
+			
+			$sqlservername = $server.ConnectionContext.ExecuteScalar("select @@servername")
+			$instance = $server.InstanceName
+			
+			if ($instance.length -eq 0)
+			{
+				$serverinstancename = $server.NetName
+				$instance = "MSSQLSERVER"
+			}
+			else
+			{
+				$netname = $server.NetName
+				$serverinstancename = "$netname\$instance"
+			}
+			
+			$serverinfo = [PSCustomObject]@{
+				ServerInstanceName = $serverinstancename
+				SqlServerName = $sqlservername
+				IsEqual = $serverinstancename -eq $sqlservername
+				RenameRequired = $serverinstancename -ne $sqlservername
+			}
+			
+			if ($Detailed)
+			{
+				$reasons = @()
+				$servicename = "SQL Server Reporting Services ($instance)"
+				
+				try
+				{
+					$rs = Get-Service -ComputerName $server.ComputerNamePhysicalNetBIOS -DisplayName $servicename -ErrorAction SilentlyContinue
+				}
+				catch
+				{
+					if ($NoWarnings -eq $false)
+					{
+						Write-Warning "Can't contact $($server.ComputerNamePhysicalNetBIOS) using Get-Service. This means the script will not be able to automatically restart SQL services."
+					}
+				}
+				
+				if ($rs.count -gt 0)
```

Comment:
Something funky is going on at this line which causes bogus warnings about needing to restart SSRS, even on servers where SSRS is not installed.